openmdao/core/group.py

```diff
@@ -1374,6 +1374,34 @@ def _setup_transfers(self, recurse=True):
         if self._conn_discrete_in2out:
             self._vector_class.TRANSFER._setup_discrete_transfers(self, recurse=recurse)
 
+    def promotes(self, subsys_name, any=None, inputs=None, outputs=None):
+        """
+        Promote a variable in the model tree.
+
+        Parameters
+        ----------
+        subsys_name : str
+            The name of the child subsystem whose inputs/outputs are being promoted.
+        any : Sequence of str or tuple
+            A Sequence of variable names (or tuples) to be promoted, regardless
+            of if they are inputs or outputs. This is equivalent to the items
+            passed via the `promotes=` argument to add_subsystem.  If given as a
+            tuple, we use the "promote as" standard of ('real name', 'promoted name')*[]:
+        inputs : Sequence of str or tuple
+            A Sequence of input names (or tuples) to be promoted. Tuples are
+            used for the "promote as" capability.
+        outputs : Sequence of str or tuple
+            A Sequence of output names (or tuples) to be promoted. Tuples are
+            used for the "promote as" capability.
+        """
+        subsys = getattr(self, subsys_name)
+        if any:
+            subsys._var_promotes['any'] = any
```

Comment:
This looks like it would replace any existing values in the _var_promotes dictionary.  I think it is more correct to append them to the existing dictionary instead, in case the user want to call `promotes` twice on the same subsystem.

Reply:
Fixed